$OpenBSD: patch-src_process_ProcessTree_c,v 1.1 2018/05/01 08:43:28 landry Exp $

Backport https://bitbucket.org/tildeslash/monit/commits/4a48ff1f9adaa53946492b512347ebdf85fee3ee to avoid trigerring monit: vfprintf %s NULL in "%s" in syslog

Index: src/process/ProcessTree.c
--- src/process/ProcessTree.c.orig
+++ src/process/ProcessTree.c
@@ -304,7 +304,7 @@ boolean_t ProcessTree_updateProcess(Service_T s, pid_t
                 s->inf.process->threads           = ptree[leaf].threads.self;
                 s->inf.process->children          = ptree[leaf].children.total;
                 s->inf.process->zombie            = ptree[leaf].zombie;
-                snprintf(s->inf.process->secattr, STRLEN, "%s", ptree[leaf].secattr);
+                snprintf(s->inf.process->secattr, STRLEN, "%s", NVLSTR(ptree[leaf].secattr));
                 if (ptree[leaf].cpu.usage.self >= 0) {
                         // compute only if initialized (delta between current and previous snapshot is available)
                         s->inf.process->cpu_percent = _cpuUsage(ptree[leaf].cpu.usage.self, ptree[leaf].threads.self);
